### PR TITLE
e2e longevity test

### PR DESCRIPTION
**Pull request by vishnoianil**
**- What this PR does and why is it needed**
PR adds an e2e test to ensure that the cluster is able to handle creating containers continuously for 20 minutes or so. It is meant to run on the master branch only due to its time requirement (but currently runs on all branches for easier testing).

Signed-off-by: Andrew Sun <asun@redhat.com>
Signed-off-by: Anil Vishnoi <vishnoianil@gmail.com>

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -1653,3 +1653,102 @@ func getNodePodCIDR(nodeName string) (string, error) {
...
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	framework.Logf("Created pod %q with IP %q %q", result.GetName(), result.Status.PodIP, result.Status.PodIPs)
+
+	err = e2epod.WaitForPodSuccessInNamespace(f.ClientSet, podName, f.Namespace.Name)
+
+	if err != nil {
```
Comment (aojea) [inline test/e2e/e2e_test.go:1706]: shouldn´t we return here instead of continue and trying to delete the pod?

Comment (vishnoianil) [reply test/e2e/e2e_test.go:1706]: Yup.Thanks for catching. Fixing in upcoming patch.

**Comment by vishnoianil** [comment]
@aojea @nerdalert @abhat PTAL, Thanks

File: test/e2e/e2e_test.go [hunk 2]
```diff
@@ -1653,3 +1705,68 @@ func getNodePodCIDR(nodeName string) (string, error) {
 	}
 	return "", fmt.Errorf("could not parse annotation %q", annotation)
 }
+
+func checkPodSleep(f *framework.Framework, nodeName, podName string, errChan chan error) {
+	//	defer GinkgoRecover()
```
Comment (nerdalert) [inline test/e2e/e2e_test.go:1710]: did you want this commented out?

**Review by nerdalert — COMMENTED** [review]
I think the Ginkgo framework timeout after 300 seconds/5 minutes. Not sure how to wait in a test for 20 minutes without the test failing at the 5 minute mark. ty for adding to e2e!

File: test/e2e/e2e_test.go (hunk 2 above)
Comment (vishnoianil) [reply test/e2e/e2e_test.go:1710]: I need to remove this. Will fix it.

**Comment by vishnoianil** [comment]
> I think the Ginkgo framework timeout after 300 seconds/5 minutes. Not sure how to wait in a test for 20 minutes without the test failing at the 5 minute mark. ty for adding to e2e!

@nerdalert Test do run for 20 minutes ( i can see in the logs). Timeout will happen only if the pods are stuck and didn't go through the lifecycle ( create, ready, get-ip, run-command, success). Or am i missing something here?

File: test/e2e/e2e.go [hunk 3]
```diff
@@ -1659,6 +1659,58 @@ var _ = ginkgo.Describe("e2e ingress gateway traffic validation", func() {
...
+		for time.Since(startTime) < 20*time.Minute {
+			// spin up 10 containers that each do nothing for 30 seconds
+
+			var errChans [10]chan error
+			for i, _ := range errChans {
+				errChans[i] = make(chan error)
+				go checkPodSleep(f, "", fmt.Sprintf("test-container-%v", pod), errChans[i])
+				pod++
+
+				time.Sleep(500 * time.Millisecond)
```
Comment (jluhrsen) [inline test/e2e/e2e.go:1698]: what's this for?

File: test/e2e/e2e.go [hunk 4]
```diff
@@ -1659,6 +1659,58 @@ var _ = ginkgo.Describe("e2e ingress gateway traffic validation", func() {
 	})
 })
 
+// This test perform a longevity test, where it spawns multiple container
+// and wait for them to finish executing sleep. It does this exercise for
+// 20 minutes and stops. Each container starts, sleeps for 10 seconds and
+// goes to success or failure state.
```
Comment (jluhrsen) [inline test/e2e/e2e.go:1665]: nice simple longevity test :)

I think a cool extension to this (maybe I can take it up at some point) would be to add some network activity from
each pod that comes up to "somewhere" (another pod, service, node, etc) and then parse the pod log to make
sure that was successful. Could catch some corner cases or races. just thinking out loud.

File: test/e2e/e2e.go (hunk 3 above)
Comment (vishnoianil) [reply test/e2e/e2e.go:1698]: @jluhrsen sleep time between pod creation.

File: test/e2e/e2e.go (hunk 4 above)
Comment (vishnoianil) [reply test/e2e/e2e.go:1665]: @jluhrsen We can run iperf client/server in these pods and log/evaluate the result. That would be good improvement, but i am not sure if github infra will give us enough resources to run the test with some level of stability.

The ideal behind this test was to give a bit of sanity check for upstream patches. It creates around 600-700 pods in this test, so atleast it will be helpful to surface any low hanging bugs. But by no means this test is enough. We should have a proper longevity test with more chaos tests running to really check the stability of the code.

File: test/e2e/e2e.go (hunk 3 above)
Comment (jluhrsen) [reply test/e2e/e2e.go:1890]: I know :)

I should have ask why you have it.

File: test/e2e/e2e.go (hunk 4 above)
Comment (jluhrsen) [reply test/e2e/e2e.go:1857]: agreed. I'll add a story to my teams back-log so I can pick it up at some point.

File: test/e2e/e2e.go (hunk 3 above)
Comment (vishnoianil) [reply test/e2e/e2e.go:1890]: Just to avoid choking github low spec VM with sudden cpu/mem spike. :)
